react-backend/app.js

```diff
@@ -4,43 +4,37 @@ const favicon = require('serve-favicon');
 const logger = require('morgan');
 const cookieParser = require('cookie-parser');
 const bodyParser = require('body-parser');
-const apiRoutes = require('./router');
-
-const app = express();
-
-app.use(favicon(path.join(__dirname, 'public', 'favicon.ico')));
-app.use(logger('dev'));
-app.use(bodyParser.json());
-app.use(bodyParser.urlencoded({ extended: true }));
-app.use(cookieParser());
-app.use(express.static(path.join(__dirname, 'client/build')));
-
-app.use('/api', apiRoutes);
-app.get('/*', express.static(path.join(__dirname, 'client/build/index.html')));
-
-// catch 404 and forward to error handler
-app.use((req, res, next) => {
-  const err = new Error('Not Found');
-  err.status = 404;
-  next(err);
-});
-
-// error handler
-app.use((err, req, res) => {
-  if (err === 'notauthorized') {
-    return res.status(401).send({ error: 'Not logged in' });
-  }
-  // set locals, only providing error in development
-  const message = err.message;
-  const error = req.app.get('env') === 'development' ? err : {};
-
-  // render the error page
-  res.status(err.status || 500);
-  res.send({ error, message });
-});
-
-const port = process.env.NODE_ENV === 'test' ? 5010 : process.env.PORT || 5000;
-
-app.listen(port);
-
-module.exports = app;
+const postmark = require('postmark');
+const createPgPool = require('./database/dbConnectionLazy');
+const { notFound, errorHandler } = require('./shared/middleware');
+
+
+module.exports = (cfg) => {
+  // Non-global import required because of client-connections
+  // that are created in module scope of routes
+  const apiRoutes = require('./router'); // eslint-disable-line global-require
+
+  const postmarkClient = new postmark.Client(cfg.email.postmark_key);
+  const psqlClient = createPgPool(cfg);
+
+  const app = express();
+
+  app.set('cfg', cfg);
+  app.set('client:postmark', postmarkClient);
+  app.set('client:psql', psqlClient);
```

Comment:
This feels a bit gross, why are we doing this? 🙈

Reply:
Why is it gross?

Storing the clients on the app object means they can be accessed from the handlers without needing to require anything in. That way they can be passed as arguments to functions that need them.

Reply:
It feels like a weird separation of concerns thing having access on each route. Would we be accessing the dB from the request from now on?

Reply:
Not sure what you mean by "weird separation of concerns thing", nor exactly what the objection to this is, but we can talk about it during our chat tomorrow.

Yes, handlers would access database clients via `req.app` and then pass it on as an argument to whatever functions needed it.

Reply:
Think of it this way: before, handlers and functions below them in the hierarchy were relying on object references hauled in from outside using `require`. Now, they only rely on their arguments.

Reply:
I'm not objecting in the sense that I don't think it should be merged, just that it seems strange compared to the _limited setups worked with_. All the other config middleware you've added seems to add variables that we can access them where needed, whereas this is adding a db pool just incase we need it. 

I guess I don't really see where the separation is from what should be added to the request through middleware. Technically we could add all our functions as middleware too and only ever require things once, but that would be mental...

Also  😭,updating Travs envs has broken my tests, so let's get this merged first 😊.

Reply:
> Also 😭,updating Travs envs has broken my tests

Really sorry, I wasn't thinking when I did that. You can just add your env vars back to the config, and it should work.

> whereas this is adding a db pool just incase we need it

Ah, I think there's been  a misunderstanding. I'm not adding the pool just in case we need it. We _will_ need it. Just not in this PR. See my PR description:

> This is actually only a partial update because currently dbConnection.js and all files relying on it are unchanged. I decided to leave this because it would inflate the diff unnecessarily. I will finish off this work in a subsequent PR.

In the next PR, I'm going to remove all references to `dbConnection.js`.